views/campground_requests.py

```diff
@@ -0,0 +1,63 @@
+import sqlite3
+from models import Campground
+
+def get_all_campgrounds():
+    """Gets all campgrounds
+    Returns:
+        dict: All campground dictionaries"""
+    with sqlite3.connect("./national_park.sqlite3") as conn:
+        conn.row_factory = sqlite3.Row
+        db_cursor = conn.cursor()
+
+        db_cursor.execute("""
+        SELECT
+            c.id,
+            c.name,
+            c.park_id,
+            c.available_sites,
+            c.description
+        FROM Campground c
+        """)
+
+        campgrounds = []
+
+        dataset = db_cursor.fetchall()
+
+        for row in dataset:
+
+            campground = Campground(row['id'], row['name'], row['park_id'], row['available_sites'], row['description'])
+
+            campgrounds.append(campground.__dict__)
+
+    return campgrounds
+
+def get_single_campground(id):
+    """Finds the matching campground dictionary for the specified campground id
+
+    Args:
+        id (int): campground id
+
+    Returns:
+        object: campground dictionary
+    """
+    with sqlite3.connect("./national_park.sqlite3") as conn:
+        conn.row_factory = sqlite3.Row
```

Comment:
Each of the "all" and "single" functions use the same code over an over again. What are your thoughts on making two, new functions in this module with more general names like `get_all` and `get_single`.

* Each would take in the SQL string as a parameter.
* The single function would return a dictionary
* The all function would return a list of dictionaries

Reply:
Example....

```py
def get_single(sql):
    with sqlite3.connect("./national_park.sqlite3") as conn:
        conn.row_factory = sqlite3.Row
        db_cursor = conn.cursor()
        db_cursor.execute(sql)

        row_dict = db_cursor.fetchone()
        return row_dict
```

Reply:
Great feedback, we decided to incorporate your advice.  We added a sql_helper module to make reusable get_single and get_all functions.